test/e2e/rollback_byo_test.go

```diff
@@ -0,0 +1,174 @@
+	if err != nil {
+		t.Fatalf("Failed to create initial Service: %v: %v", names.Service, err)
+	}
+	url := resources.Route.Status.URL.URL()
+	if _, err = pkgTest.WaitForEndpointState(
+		clients.KubeClient,
+		t.Logf,
+		url,
+		v1a1test.RetryingRouteInconsistency(pkgTest.MatchesAllOf(pkgTest.IsStatusOK, pkgTest.MatchesBody(test.HelloWorldText))),
+		"HelloWorldServesText",
+		test.ServingFlags.ResolvableDomain); err != nil {
+		t.Fatalf("The endpoint %s for Route %s didn't serve the expected text %q: %v", url, names.Route, test.HelloWorldText, err)
+	}
+
+	revisionName := resources.Revision.ObjectMeta.Name
+	if revisionName != byoNameOld {
+		t.Fatalf("Expect configuration name in revision label %q but got %q ", byoNameOld, revisionName)
+	}
+
+	// Update service to use a new byo name
+	t.Logf("Updating the Service to a new revision with a new byo name %q.", byoNameNew)
+	newSvc := resources.Service.DeepCopy()
+	so := WithInlineConfigSpec(*v1a1test.ConfigurationSpec(pkgTest.ImagePath(test.PizzaPlanet1), &byoNameNew))
+	so(newSvc)
+	withTrafficSpecNew(newSvc)
+	svc, err := v1a1test.PatchService(t, clients, resources.Service, newSvc)
+	resources.Service = svc
+	if err != nil {
+		t.Fatalf("Patch update for Service (new byo name %q) failed: %v", byoNameNew, err)
+	}
+
+	t.Log("Since the Service was updated a new Revision will be created and the Service will be updated")
+	newRevision, err := v1a1test.WaitForServiceLatestRevision(clients, names)
+	if err != nil {
+		t.Fatalf("Service %s was not updated with the Revision for new byo name %s: %v", names.Service, byoNameNew, err)
+	}
+	if newRevision != byoNameNew {
+		t.Fatalf("Expect configuration name in revision label %q but got %q ", byoNameNew, newRevision)
+	}
+
+	// Now, rollback to the first RevisionSpec
+	rollbackSvc := resources.Service.DeepCopy()
+	rollbackSvc.Spec = v1alpha1.ServiceSpec{
+		DeprecatedRelease: &v1alpha1.ReleaseType{
+			Revisions:      []string{byoNameOld, byoNameNew},
+			RolloutPercent: 0,
+			Configuration: v1alpha1.ConfigurationSpec{
+				Template: &v1alpha1.RevisionTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: byoNameNew,
+					},
+					Spec: v1alpha1.RevisionSpec{
+						DeprecatedContainer: &corev1.Container{
+							Image: pkgTest.ImagePath(test.PizzaPlanet1),
+						},
+					},
+				},
+			},
+		},
+	}
+	svc, err = v1a1test.PatchService(t, clients, resources.Service, rollbackSvc)
+	resources.Service = svc
+	if err != nil {
+		t.Fatalf("Patch update for Service (rollback to byo name %q) failed: %v", byoNameOld, err)
+	}
+
+	t.Logf("We are rolling back to the previous revision (byoNameOld %q).", byoNameOld)
+	// Wait for the route to become ready, and check that the traffic split between the byoNameOld
+	// and byoNameNew is 100 and 0, respectively
+	err = v1a1test.WaitForServiceState(clients.ServingAlphaClient, names.Service, func(s *v1alpha1.Service) (bool, error) {
```

Comment:
```suggestion
	if err = v1a1test.WaitForServiceState(clients.ServingAlphaClient, names.Service, func(s *v1alpha1.Service) (bool, error) {}, "ServiceRollbackRevision"); err != nil 
```

Reply:
Thanks for the suggestion! I thought about doing a one liner like you suggested, but decided not to for readability reasons because of the function definition inside the call